examples/blog-apollo-server/src/executor.ts

```diff
@@ -0,0 +1,33 @@
+import { GraphQLExecutor } from "apollo-server-core";
+import { GraphQLSchema } from "graphql";
+import {
+  CompiledQuery,
+  compileQuery,
+  CompilerOptions,
+  isCompiledQuery
+} from "graphql-jit";
+import LRU from "tiny-lru";
+
+export const executor = (
+  schema: GraphQLSchema,
+  cacheSize = 1024,
+  compilerOpts: Partial<CompilerOptions> = {}
+): GraphQLExecutor => {
+  const cache = LRU<CompiledQuery>(cacheSize);
+  return async ({ context, document, operationName, request, queryHash }) => {
+    const prefix = operationName || "NotParametrized";
+    const cacheKey = `${prefix}-${queryHash}`;
+    let compiledQuery = cache.get(cacheKey);
+    if (!compiledQuery) {
+      const compilationResult = compileQuery(schema, document, operationName || undefined, compilerOpts);
```

Comment:
```suggestion
      const compilationResult = compileQuery(schema, document, operationName, compilerOpts);
```

Is there a valid case where ` || undefined` is useful?

Reply:
Unfortunately apollo defines operationName as `string | null` and graphql-jit requires `string | undefined`. So I need this extra `null => undefined` conversion to compile.

Reply:
@boopathi we should extend what we expect. We dont care if it is null or undefined.